code/core/src/preview/runtime.ts

```diff
@@ -6,25 +6,30 @@ import { globalPackages, globalsNameReferenceMap } from './globals/globals';
 import { globalsNameValueMap } from './globals/runtime';
 import { prepareForTelemetry } from './utils';
 
-// Apply all the globals
-globalPackages.forEach((key) => {
-  (global as any)[globalsNameReferenceMap[key]] = globalsNameValueMap[key];
-});
+export function setup() {
+  // Apply all the globals
+  globalPackages.forEach((key) => {
+    (global as any)[globalsNameReferenceMap[key]] = globalsNameValueMap[key];
+  });
 
-global.sendTelemetryError = (error: any) => {
-  const channel = global.__STORYBOOK_ADDONS_CHANNEL__;
-  channel.emit(TELEMETRY_ERROR, prepareForTelemetry(error));
-};
+  global.sendTelemetryError = (error: any) => {
+    const channel = global.__STORYBOOK_ADDONS_CHANNEL__;
+    channel.emit(TELEMETRY_ERROR, prepareForTelemetry(error));
+  };
 
-// handle all uncaught StorybookError at the root of the application and log to telemetry if applicable
-global.addEventListener('error', (args: any) => {
-  const error = args.error || args;
-  if (error.fromStorybook) {
-    global.sendTelemetryError(error);
-  }
-});
-global.addEventListener('unhandledrejection', ({ reason }: any) => {
-  if (reason.fromStorybook) {
-    global.sendTelemetryError(reason);
-  }
-});
+  // handle all uncaught StorybookError at the root of the application and log to telemetry if applicable
+  global.addEventListener('error', (args: any) => {
+    const error = args.error || args;
+    if (error.fromStorybook) {
+      global.sendTelemetryError(error);
+    }
+  });
+  global.addEventListener('unhandledrejection', ({ reason }: any) => {
+    if (reason.fromStorybook) {
+      global.sendTelemetryError(reason);
+    }
+  });
+}
+
+// TODO: In the future, remove this call to make the module side-effect free
+setup();
```

Comment:
Are you not calling this function twice now? Can we not just remove it, now we call it explicitly in codegen-modern-iframe-script.ts?

Reply:
Okay, I will remove this call, let me know if you have objections

Reply:
I see, we can not remove it because of webpack.